go-controller/pkg/ovn/ovn.go

```diff
@@ -558,25 +558,26 @@ func (oc *Controller) WatchPods() {
 		UpdateFunc: func(old, newer interface{}) {
 			oldPod := old.(*kapi.Pod)
 			pod := newer.(*kapi.Pod)
-			if !podWantsNetwork(pod) {
-				if oldPod.Annotations[routingNamespaceAnnotation] != pod.Annotations[routingNamespaceAnnotation] ||
-					oldPod.Annotations[routingNetworkAnnotation] != pod.Annotations[routingNetworkAnnotation] {
-					oc.deletePodExternalGW(oldPod)
-					if err := oc.addPodExternalGW(pod); err != nil {
-						klog.Errorf(err.Error())
-					}
-				}
-				return
-			}
 
 			_, retry := retryPods.Load(pod.UID)
-			if podScheduled(pod) && retry {
+			if podScheduled(pod) && retry && podWantsNetwork(pod) {
 				if err := oc.addLogicalPort(pod); err != nil {
 					klog.Errorf(err.Error())
 					oc.recordPodEvent(err, pod)
 				} else {
 					retryPods.Delete(pod.UID)
 				}
+			} else {
+				// No matter if a pod is ovn networked, or host networked, we still need to check for exgw
+				// annotations. If the pod is ovn networked and is in update reschedule, addLogicalPort will take
+				// care of updating the exgw updates
```

Comment:
ok, although it seems like it might be simpler at this point to _not_ call `addPodExternalGW` from `addLogicalPort` and always do it from here?

Reply:
probably, but that's a bigger change that I would prefer holding off on